java-client/src/main/java/com/wavefront/integrations/Wavefront.java

```diff
@@ -142,6 +143,21 @@ private void internalSend(String name, double value, @Nullable Long timestamp, S
         // already connected.
       }
     }
+    try {
+      internalSend0(name, value, timestamp, source, pointTags);
+    } catch (SocketException ex) { // retry at least once on SocketExceptions
+      this.socket = null;
+      try {
+        connect();
+      } catch (IllegalStateException e) {
+        // already connected.
```

Comment:
you should probably log both the SocketException and the IllegalStateException.

Even if you're pretty sure you only get IllegalStateException iff "already connected", probably a good idea to log just to be sure